Python PR checks are flaky

Sometimes the Python PR checks fail due to a conflict in environment name. Usually passes on rerun. Example [run](https://github.com/pulumi/esc-sdk/actions/runs/11356678743/job/31588295010?pr=40)

```
=================================== FAILURES ===================================
___________________ TestEscApi.test_check_environment_valid ____________________

self = <test.test_esc_api.TestEscApi testMethod=test_check_environment_valid>

    def tearDown(self) -> None:
        if self.baseEnvName is not None:
>           self.client.delete_environment(self.orgName, PROJECT_NAME, self.baseEnvName)

test/test_esc_api.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pulumi_esc_sdk/esc_client.py:171: in delete_environment
    self.esc_api.delete_environment(org_name, project_name, env_name)
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/pydantic/validate_call_decorator.py:60: in wrapper_function
    return validate_call_wrapper(*args, **kwargs)
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/pydantic/_internal/_validate_call.py:96: in __call__
    res = self.__pydantic_validator__.validate_python(pydantic_core.ArgsKwargs(args, kwargs))
pulumi_esc_sdk/api/esc_api.py:1690: in delete_environment
    return self.api_client.response_deserialize(
pulumi_esc_sdk/api_client.py:322: in response_deserialize
    raise ApiException.from_response(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'pulumi_esc_sdk.exceptions.ApiException'>

    @classmethod
    def from_response(
        cls,
        *,
        http_resp,
        body: Optional[str],
        data: Optional[Any],
    ) -> Self:
        if http_resp.status == 400:
            raise BadRequestException(http_resp=http_resp, body=body, data=data)
    
        if http_resp.status == 401:
            raise UnauthorizedException(http_resp=http_resp, body=body, data=data)
    
        if http_resp.status == 403:
            raise ForbiddenException(http_resp=http_resp, body=body, data=data)
    
        if http_resp.status == 404:
            raise NotFoundException(http_resp=http_resp, body=body, data=data)
    
        if 500 <= http_resp.status <= 599:
            raise ServiceException(http_resp=http_resp, body=body, data=data)
>       raise ApiException(http_resp=http_resp, body=body, data=data)
E       pulumi_esc_sdk.exceptions.ApiException: (409)
E       Reason: Conflict
E       HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json', 'Content-Length': '118', 'Connection': 'keep-alive', 'Date': 'Wed, 16 Oct 2024 00:38:[27](https://github.com/pulumi/esc-sdk/actions/runs/11356678743/job/31588295010?pr=40#step:5:29) GMT', 'X-Pulumi-Request-Id': '2b136c66-8460-41af-ac12-45aafd8c011a', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 6ee3eecd683392[28](https://github.com/pulumi/esc-sdk/actions/runs/11356678743/job/31588295010?pr=40#step:5:30)6f206a7ea6e9ac0c.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'SFO53-P2', 'X-Amz-Cf-Id': 'xoJLjlW0tlM5wpn0J8lCtCKdRVvStxmYSLdjTihSlJj64nlsWxgkBg==', 'Referrer-Policy': 'strict-origin-when-cross-origin', 'Strict-Transport-Security': 'max-age=[31](https://github.com/pulumi/esc-sdk/actions/runs/11356678743/job/31588295010?pr=40#step:5:33)536000; includeSubDomains; preload'})
E       HTTP response body: {"code":409,"message":"Conflict: a environment with name \"sdk-python-test/env-base-20241016003827\" already exists"}

pulumi_esc_sdk/exceptions.py:157: ApiException
=========================== short test summary info ============================
FAILED test/test_esc_api.py::TestEscApi::test_check_environment_valid - pulum...
========================= 1 failed, 2 passed in 5.51s ==========================
make: *** [Makefile:56: test_python] Error 1
```

--- a/sdk/python/test/test_esc_api.py
+++ b/sdk/python/test/test_esc_api.py
@@ -9,7 +9,7 @@ from datetime import datetime
 import pulumi_esc_sdk as esc
 
 PROJECT_NAME = "sdk-python-test"
-ENV_PREFIX = "env-"
+ENV_PREFIX = "env"
 
 
 class TestEscApi(unittest.TestCase):
@@ -27,7 +27,7 @@ class TestEscApi(unittest.TestCase):
 
         self.remove_all_python_test_envs()
 
-        self.baseEnvName = ENV_PREFIX + "base-" + datetime.now().strftime("%Y%m%d%H%M%S")
+        self.baseEnvName = f"{ENV_PREFIX}-base-{datetime.now().timestamp()}"
         self.client.create_environment(self.orgName, PROJECT_NAME, self.baseEnvName)
         self.envName = None
 
@@ -38,7 +38,7 @@ class TestEscApi(unittest.TestCase):
             self.client.delete_environment(self.orgName, PROJECT_NAME, self.envName)
 
     def test_environment_end_to_end(self) -> None:
-        self.envName = ENV_PREFIX + "end-to-end-" + datetime.now().strftime("%Y%m%d%H%M%S")
+        self.envName = f"{ENV_PREFIX}-end-to-end-{datetime.now().timestamp()}"
         self.client.create_environment(self.orgName, PROJECT_NAME, self.envName)
 
         envs = self.client.list_environments(self.orgName)

